remote/remote_test.go

```diff
@@ -1027,6 +1042,33 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 				h.AssertEq(t, remoteLabel, "newValue")
 			})
 
+			it("can be pulled from unauthenticated registry", func() {
+				img, err := remote.NewImage(repoName, authn.DefaultKeychain)
+				h.AssertNil(t, err)
+
+				err = img.SetLabel("mykey", "newValue")
+				h.AssertNil(t, err)
+
+				h.AssertNil(t, img.Save())
+
+				name, err := name.ParseReference(repoName, name.WeakValidation)
```

Comment:
I'm a bit confused... how does this work? It looks like `repoName` is on the authenticated registry. Where are we pulling from the unauthenticated one?

Reply:
Yeah, this could use some clarification. What it attempts to do is:
1. Write a new image on the auth'd registry with ref like: `localhost:5555/my-image-1234`
2. Parse out the name out of the auth'd ref: `my-image.1234`
3. Rebuild the ref string for the same image on the unauth'd repo: `localhost:6666/my-image-1234`
4. Show that we can pull that without any credentials (since we never added any for `localhost:6666` to the `DOCKER_CONFIG`).

Reply:
I'll take another pass and see if I can clean it up. TBH, the test is mostly here for some coverage for our `WithSharedStorageVolume` helper, theoretically unauthenticated registries are covered with the DockerHub examples.